superset/assets/src/views/chartList/ChartList.tsx

```diff
@@ -0,0 +1,316 @@
+          <span className='no-wrap'>{moment(changed_on).fromNow()}</span>
+        ),
+      Header: t('Last Modified'),
+      accessor: 'changed_on',
+      sortable: true,
+    },
+    {
+      Cell: ({ row: { state, original } }: any) => {
+        const handleDelete = () => this.handleChartDelete(original);
+        const handleEdit = () => this.handleChartEdit(original);
+        if (!this.canEdit && !this.canDelete) {
+          return null;
+        }
+
+        return (
+          <span className={`actions ${state && state.hover ? '' : 'invisible'}`}>
+            {this.canDelete && (
+              <ConfirmStatusChange
+                title={t('Please Confirm')}
+                description={<>{t('Are you sure you want to delete')} <b>{original.slice_name}</b>?</>}
+                onConfirm={handleDelete}
+              >
+                {(confirmDelete) => (
+                  <span
+                    role='button'
+                    className='action-button'
+                    onClick={confirmDelete}
+                  >
+                    <i className='fa fa-trash' />
+                  </span>
+                )}
+              </ConfirmStatusChange>
+            )}
+            {this.canEdit && (
+              <span
+                role='button'
+                className='action-button'
+                onClick={handleEdit}
+              >
+                <i className='fa fa-pencil' />
+              </span>
+            )}
+          </span>
+        );
+      },
+      Header: 'Actions',
+      id: 'actions',
+    },
+  ];
+
+  public hasPerm = (perm: string) => {
+    if (!this.state.permissions.length) {
+      return false;
+    }
+
+    return Boolean(this.state.permissions.find((p) => p === perm));
+  }
+
+  public handleChartEdit = ({ id }: { id: number }) => {
+    window.location.assign(`/chart/edit/${id}`);
+  }
+
+  public handleChartDelete = ({ id, slice_name }: Chart) => {
+    SupersetClient.delete({
+      endpoint: `/api/v1/chart/${id}`,
+    }).then(
+      (resp) => {
+        const { lastFetchDataConfig } = this.state;
+        if (lastFetchDataConfig) {
+          this.fetchData(lastFetchDataConfig);
+        }
+        this.props.addSuccessToast(t('Deleted') + ` ${slice_name}`);
+      },
+      (err: any) => {
+        this.props.addDangerToast(t('There was an issue deleting') + `${slice_name}`);
+      },
+    );
+  }
+  public handleBulkDashboardDelete = (charts: Chart[]) => {
+    SupersetClient.delete({
+      endpoint: `/api/v1/dashboard/?q=!(${charts.map(({ id }) => id).join(',')})`,
+    }).then(
+      ({ json = {} }) => {
+        const { lastFetchDataConfig } = this.state;
+        if (lastFetchDataConfig) {
+          this.fetchData(lastFetchDataConfig);
+        }
+        this.props.addSuccessToast(json.message);
+      },
+      (err: any) => {
+        console.error(err);
+        this.props.addDangerToast(t('There was an issue deleting the selected dashboards'));
```

Comment:
```suggestion
        this.props.addDangerToast(t(`There was an issue deleting the selected ${charts.length > 1 ? 'dashboards' : 'dashboard'}`));
```
^ This is probably not the right way to do it with translation factored in, but it's possible to select 1 item with bulk actions, so we may want to get the pluralization right.